src/mono/wasm/debugger/DebuggerTestSuite/Tests.cs

```diff
@@ -1850,6 +1851,70 @@ JObject FindFrame(JObject pause_location, string function_name)
                     ?.Where(f => f["functionName"]?.Value<string>() == function_name)
                     ?.FirstOrDefault();
 
+        [Fact]
+        public async Task DebugLazyLoadedAssemblyWithPdb()
+        {
+            var insp = new Inspector();
+            var scripts = SubscribeToScripts(insp);
+            await Ready();
+            await insp.Ready(async (cli, token) =>
+            {
+                var source_location = "dotnet://lazy-debugger-test.dll/lazy-debugger-test.cs";
+
+                ctx = new DebugTestContext(cli, insp, token, scripts);
+
+                // Simulate loading an assembly into the framework
+                byte[] bytes = File.ReadAllBytes("../../../lazy-debugger-test/wasm/Debug/lazy-debugger-test.dll");
+                string asm_base64 = Convert.ToBase64String(bytes);
+                bytes = File.ReadAllBytes("../../../lazy-debugger-test/wasm/Debug/lazy-debugger-test.pdb");
+                string pdb_base64 = Convert.ToBase64String(bytes);
+                var load_assemblies = JObject.FromObject(new
+                {
+                    expression = $"{{ let asm_b64 = '{asm_base64}'; let pdb_b64 = '{pdb_base64}'; invoke_static_method('[debugger-test] LoadDebuggerTest:LoadLazyAssembly', asm_b64, pdb_b64); }}"
+                });
+                var load_assemblies_res = await cli.SendCommand("Runtime.evaluate", load_assemblies, token);
+                Assert.True(load_assemblies_res.IsOk);
+
+                Assert.Contains(source_location, scripts.Values);
+
+                await SetBreakpoint(source_location, 9, 8);
+
+                var pause_location = await EvaluateAndCheck(
+                   "window.setTimeout(function () { invoke_static_method('[lazy-debugger-test] LazyMath:IntAdd', 5, 10); }, 1);",
+                   source_location, 9, 8,
+                   "IntAdd");
+                var locals = await GetProperties(pause_location["callFrames"][0]["callFrameId"].Value<string>());
+                CheckNumber(locals, "a", 5);
+                CheckNumber(locals, "b", 10);
+            });
+        }
+
+        [Fact]
+        public async Task DebugLazyLoadedAssemblyWithoutPdb()
```

Comment:
Please add a test with an assembly with embedded ppdb data.

Reply:
Working on it.

Reply:
Done